File: app/src/main/java/com/innso/mobile/api/config/ApiConfig.java
package com.innso.mobile.api.config;


import android.content.Context;

import com.innso.mobile.R;
import com.innso.mobile.managers.preferences.PrefsManager;

public class ApiConfig {

    public static final String BEARER = "Bearer ";
    public static final String AUTHORIZATION = "authorization";
    public static final String PARAM_AUTHORIZATION = "auth";

    public boolean DEBUG = false;

    private PrefsManager prefsManager;
    private Context context;

    public ApiConfig(Context context, PrefsManager prefsManager) {
        this.prefsManager = prefsManager;
        this.context = context;
    }

    public String getFirebaseUrl() {
        return context.getString(R.string.firebase_database);
    }

    public String getFuntionsUrlBase() {
        return "https://us-central1-innso-mobile.cloudfunctions.net";
    }
}

File: app/src/main/java/com/innso/mobile/api/config/ApiConfig.kt
package com.innso.mobile.api.config


import android.content.Context
import com.innso.mobile.BuildConfig
import com.innso.mobile.R
import javax.inject.Inject
import javax.inject.Singleton


const val BEARER = "Bearer "
const val AUTHORIZATION = "authorization"
const val PARAM_AUTHORIZATION = "auth"

@Singleton
class ApiConfig @Inject constructor(private val context: Context) {

    var DEBUG = true

    fun getFirebaseUrl(): String = context.getString(R.string.firebase_database)

    fun getFuntionsUrlBase(): String = context.getString(R.string.firebase_functions)

    fun getUrlNews(): String = context.getString(R.string.api_news)

    fun getNewsApiKey(): String = BuildConfig.newsApi
}
